platform/src/common/components/Charts/components/index.jsx

```
@@ -19,122 +22,87 @@
- * @returns {Object}
- * @description Get air quality level text, icon and color based on the value
- * @returns {Object} { airQualityText, AirQualityIcon, airQualityColor }
+ * @param {Number} value - The pollutant value.
+ * @param {String} pollutionType - The type of pollutant (e.g., 'pm2_5', 'pm10', etc.).
+ * @returns {Object} - { airQualityText, AirQualityIcon, airQualityColor }
+ * @description Get air quality level text, icon, and color based on the value.
  */
 export const getAirQualityLevelText = (value, pollutionType) => {
-  const ranges = pollutantRanges[pollutionType] || [];
+  // Handle invalid inputs
+  if (typeof value !== 'number' || isNaN(value) || value < 0) {
```

Biome

[error] isNaN is unsafe. It attempts a type coercion. Use Number.isNaN instead.

See the MDN documentation for more details.
Unsafe fix: Use Number.isNaN instead.


(lint/suspicious/noGlobalIsNan)